src/client.ts

```diff
@@ -110,37 +116,38 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
+  public subscribe(opts: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
+    // this.eventEmitter.emit('subscribe', options);
+    this.checkSubscriptionParams(opts, handler);
```

Comment:
I would just rely on the `this. applyMiddlewares` handling instead of checking it twice. This also avoids the case that sometimes the handler is called with the error and not.